docs/compute-resources.md

```diff
@@ -52,6 +52,263 @@ requirements for `Step`containers, they must be treated as if they are running i
+If users specify a Task-level resource limit, no `Step` may use more than that amount of resources.
+
+Each of these details is explained in more depth below.
+
+Some points to note:
+
+- Task-level resource requests and limits do not apply to sidecars which can be configured separately.
+- Users may not configure the Task-level and Step-level resource requirements (requests/limits)  simultaneously.
+
+### Configuring Task-level Resource Requirements
+
+Task-level resource requirements can be configured in `Task.Resources`, `TaskRun.Resources`, or `PipelineRun.TaskRunSpecs`.
+
+e.g.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: foo 
+spec:
+  steps:
+    - name: foo1
+    - name: foo2
+  resources:
+    requests:
+      cpu: 1 
+    limits:
+      cpu: 2
+```
+
+### Specifying Resource Requirement in Task
+
+Users can specify compute resources either at the Step-level or the Task-level in `Task`.
+
+To specify compute resources at the Step-level, use `Task.Step` and `Task.StepTemplate`.
+To specify compute resources at the Task-level, use `Task.Resources`.
+
+e.g.
+
+Using `Task.Resources`:
+
+```yaml
+kind: Task
+spec:
+  resources:
+    requests:
+      cpu: 2
+```
+
+Rejected when configuring `Task.Step` and `Task.Resources`
+
+```yaml
+kind: Task
+spec:
+  steps:
+    - name: foo
+      resources:
+        requests:
+          cpu: 1
+  resources:
+    requests:
+      cpu: 2
+```
+
+Rejected when configuring `Task.StepTemplate` and `Task.Resources`:
+
+```yaml
+kind: Task
+spec:
+  steps:
+    - name: foo
+  stepTemplate:
+    resources:
+      requests:
+        cpu: 1
+  resources:
+    requests:
+      cpu: 2
+```
+
+### Specifying Resource Requirement in TaskRun/PipelineRun
+
+Users can specify compute resources either at the Step-level or the Task-level in `TaskRun`.
+
+To specify compute resources at the Step-level, use `TaskRun.StepOverrides`.
+To specify compute resources at the Task-level, use `TaskRun.Resources` or `PipelineRun.TaskRunSpecs`.
+
+e.g.
+
+Using `TaskRun.Resources`:
+
+```yaml
+kind: TaskRun
+spec:
+  resources:
+    requests:
+      cpu: 2
+```
+
+Rejected when configuring `TaskRun.StepOverrides` and `TaskRun.Resources`:  
+
+```yaml
+kind: TaskRun
+spec:
+  stepOverrides:
+    - name: foo
+      resources:
+        requests:
+          cpu: 1
+  resources:
+    requests:
+      cpu: 2 
+```
+
+Rejected when configuring `PipelineRun.TaskRunSpecs.StepOverrides` and `PipelineRun.TaskRunSpecs.Resources`:  
+
+```yaml
+kind: PipelineRun
+spec:
+  taskRunSpecs:
+    - pipelineTaskName: foo
+      stepOverrides:
+        - name: foo 
+          resources:
+            requests:
+              cpu: 1 
+      resources:
+        requests:
+          cpu: 2 
+```
+
+### Specifying Resource Requirements with LimitRange
+
+User can use `LimitRange` to configure min or max compute resources for a `Task` or `Step`.
```

Comment:
Could you move this example to the section on LimitRanges? If we're going to include an example like the one you have here, it would be good to show both how it works for step-level and how it works for task-level.

Reply:
good catch. thanks this!